template.php

```diff
@@ -334,8 +334,44 @@ function megatron_preprocess_page(&$variables) {
       ));
     }
   }
-}
 
+  // Drawer nav.
+  $variables['navigation_placement'] = theme_get_setting('clf_navigation_placement');
+  $drawer_enabled = theme_get_setting('clf_navigation_placement') != 'default' && theme_get_setting('clf_navigation_placement') != 'double' && theme_get_setting('clf_navigation_placement') != 'higher';
+  $variables['drawer_enabled'] = $drawer_enabled;
+  if ($variables['main_menu'] && theme_get_setting('clf_use_primary_menu_in_drawer')) {
+    // Build links.
+    $tree = menu_tree_page_data(variable_get('menu_main_links_source', 'main-menu'));
+    $variables['main_menu'] = megatron_menu_navigation_links($tree);
+
+    // Build list.
+    $variables['page']['drawer'] = array(
+      '#theme' => 'megatron_links',
+      '#heading' => array(
+        'text' => t('Main menu'),
+        'level' => 'h2',
+        'class' => array('element-invisible'),
+      ),
+      '#links' => $variables['main_menu'],
+      '#attributes' => array(
+        'id' => 'main-menu',
+        'class' => array('nav'),
+      ),
+    );
+  }
+
+  // Add js and css for drawer option
+  if ($drawer_enabled) {
+    drupal_add_js(drupal_get_path('theme', 'megatron') . '/js/off.canvas.drawer.js');
+    drupal_add_css(drupal_get_path('theme', 'megatron') . '/css/off.canvas.drawer.css');
+  }
+
+  // Add js and css for navigation sticky option
+  if (theme_get_setting('clf_sticky_option')) {
```

Comment:
Maybe this sticky option can be a separate pull request or is it needed for this to work?